OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -179,7 +186,27 @@ public void Tick(Actor self)
 		public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
 			return world.ScreenMap.FrozenActorsInBox(owner, wr.Viewport.TopLeft, wr.Viewport.BottomRight)
-				.Where(f => f.Visible)
+				.Where(f =>
```

Comment:
Could you walk me through what this logic block is trying to solve? You've clearly engineered it very carefully to fix some of the issues you were seeing but it's difficult to see why each of these cases is needed. I'm wondering if there's something we could do to simply this concept.